packages/devtools_app/lib/src/screens/logging/logging_screen_v2/log_data.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/foundation.dart';
+
+import '../../../shared/diagnostics/diagnostics_node.dart';
+import '../../../shared/primitives/utils.dart';
+import '../../../shared/ui/search.dart';
+
+/// A log data object that includes optional summary information about whether
+/// the log entry represents an error entry, the log entry kind, and more
+/// detailed data for the entry.
+///
+/// The details can optionally be loaded lazily on first use. If this is the
+/// case, this log entry will have a non-null `detailsComputer` field. After the
+/// data is calculated, the log entry will be modified to contain the calculated
+/// `details` data.
+class LogDataV2 with SearchableDataMixin {
+  LogDataV2(
+    this.kind,
+    this._details,
+    this.timestamp, {
+    this.summary,
+    this.isError = false,
+    this.detailsComputer,
+    this.node,
+  }) {
+    // Fetch details immediately on creation.
+    unawaited(compute());
+  }
+
+  final String kind;
+  final int? timestamp;
+  final bool isError;
+  final String? summary;
+
+  final RemoteDiagnosticsNode? node;
+  String? _details;
+  Future<String> Function()? detailsComputer;
+
+  static const prettyPrinter = JsonEncoder.withIndent('  ');
+
+  String? get details => _details;
+
+  ValueListenable<bool> get detailsComputed => _detailsComputed;
+  final _detailsComputed = ValueNotifier<bool>(false);
+
+  Future<void> compute() async {
+    if (!detailsComputed.value) {
+      if (detailsComputer != null) {
+        _details = await detailsComputer!();
+      }
+      detailsComputer = null;
+      _detailsComputed.value = true;
+    }
+  }
+
+  /// The current calculated display height.
+  double? height;
+
+  /// The current offset of this log entry in the logs table.
+  double? offset;
```

Comment:
correction: these two fields were added to the LogDataV2 class since these were the values tracked in the wrapper classes that were deleted.